Коллекционируем пользователей

elka1907 opened the pull request:
Направляю.

elka1907 committed:
Коллекционируем пользователей

MarselYsup commented on src/main/java/school/faang/User.java:
```diff
@@ -0,0 +1,37 @@
+package school.faang;
+
+import java.util.*;
```
Не стоит использовать импорт всего (*). Делайте только точечные импорты, необходимые для текущего класса. Это улучшает читаемость кода и упрощает рефакторинг.
Если пользуешься idea, то он автоматически схлопывает, почитай эту статью тут написано, как убрать эту функцию https://www.baeldung.com/intellij-disable-wildcard-import

MarselYsup commented on src/main/java/school/faang/User.java:
```diff
@@ -0,0 +1,37 @@
+    public static Map<User, String> findHobbyLovers(List<User> users, Set<String> targetActivities) {
+        Map<User, String> result = new HashMap<>();
+
+        // Проходим по всем пользователям
```
не стоит писать комментарии везде(только в крайних случаях/сложной логике), обычно код должен быть написано так, что не требует комментариев -  у тебя все предельно понятно, поэтому комментарии излишне - они только засоряют код - в реальной разработке также, не стоит все комментировать, что пишешь

MarselYsup commented on src/main/java/school/faang/Main.java:
```diff
@@ -1,4 +1,47 @@
+    public static void main(String[] args) {
+
+        User user1 = new User(1, "Анна", 25,
+                new HashSet<>(Arrays.asList("чтение", "плавание", "программирование")));
```
можно проще Set.of (...)

MarselYsup commented on src/main/java/school/faang/Main.java:
```diff
@@ -1,4 +1,47 @@
+        User user4 = new User(4, "Петр", 35,
+                new HashSet<>(Arrays.asList("фотография", "путешествия")));
+
+        List<User> users = Arrays.asList(user1, user2, user3, user4);
```
Если список не меняется, то лучше использовать List.of - он более производительнее, а также делает список неизменяемый(в него больше ничего нельзя записать)

MarselYsup commented on src/main/java/school/faang/Main.java:
```diff
@@ -1,4 +1,47 @@
+
+        List<User> users = Arrays.asList(user1, user2, user3, user4);
+
+        Set<String> targetActivities = new HashSet<>(Arrays.asList(
```
можно проще Set.of

MarselYsup reviewed (commented):
Основная реализация выполнена отлично!)
За тесты и форматирование строк респект!)

elka1907 committed:
скорректировала:
1. new HashSet<>(Arrays.asList на Set.of
2. Arrays.asList на List.of
3. import java.util.*; на import java.util.HashMap; import java.util.List; import java.util.Map; import java.util.Set;
